rdma-ndd/doc/man/rdma-ndd.8.in

```diff
@@ -0,0 +1,93 @@
+.\" Man page generated from reStructuredText.
+.
+.TH RDMA-NDD 8 "@BUILD_DATE@" "" "OpenIB Diagnostics"
+.SH NAME
+RDMA-NDD \- RDMA device Node Description update daemon
+.
+.nr rst2man-indent-level 0
+.
+.de1 rstReportMargin
+\\$1 \\n[an-margin]
+level \\n[rst2man-indent-level]
+level margin: \\n[rst2man-indent\\n[rst2man-indent-level]]
+-
+\\n[rst2man-indent0]
+\\n[rst2man-indent1]
+\\n[rst2man-indent2]
+..
+.de1 INDENT
+.\" .rstReportMargin pre:
+. RS \\$1
+. nr rst2man-indent\\n[rst2man-indent-level] \\n[an-margin]
+. nr rst2man-indent-level +1
+.\" .rstReportMargin post:
+..
+.de UNINDENT
+. RE
+.\" indent \\n[an-margin]
+.\" old: \\n[rst2man-indent\\n[rst2man-indent-level]]
+.nr rst2man-indent-level -1
+.\" new: \\n[rst2man-indent\\n[rst2man-indent-level]]
+.in \\n[rst2man-indent\\n[rst2man-indent-level]]u
+..
+.SH SYNOPSIS
+.sp
+rdma\-ndd <options>
+.SH DESCRIPTION
+.sp
+rdma\-ndd is a system daemon which watches for rdma device changes and/or
+hostname changes and updates the Node Description of the rdma devices based on
+those changes.
+.SH DETAILS
+.sp
+Full operation of this daemon requires kernels which support polling of the
+procfs hostname file as well as libudev.
+.sp
+If your system does not support either of these features, the daemon will set
+the Node Descriptions at start up and then sleep forever.
+.SS Node Description format
+.sp
+The daemon uses the configuration options in the conf file [See below]
+specified.  %h and %d can be used as wildcards in that string to specify the
+dynamic use of <hostname> and <device> respectively.
+.sp
+NOTE: At startup and on new device detection the Node Description is always
+written to ensure the SM and this daemon are in sync.  Subsequent events will
+only write the Node Description on a device if it has changed.
+.SH OPTIONS
+.sp
+\fB\-\-retry_timer, \-t\fP
+Length of time to sleep when system errors occur when attempting to poll and or read the hostname from the system.
+.sp
+\fB\-\-retry_count, \-r\fP
+Number of times to attempt to retry setting of the node description on failure.  Default 0
+.sp
+\fB\-\-foreground, \-f\fP
+Run in the foreground instead of as a daemon
+.sp
+\fB\-\-pidfile <pidfile>\fP
+specify a pid file (daemon mode only)
+.SS Configuration flags
+.sp
+\fB\-\-config, \-z  <config_file>\fP Specify an alternate config file.
+.INDENT 0.0
+.INDENT 3.5
+Default: @CMAKE_INSTALL_FULL_SYSCONFDIR@/rdma\-ndd.conf
+.UNINDENT
+.UNINDENT
+.SS Debugging flags
+.sp
+\fB\-V, \-\-version\fP     show the version info.
+.SH FILES
+.SS CONFIG FILE
+.sp
+The default config file \(aq@CMAKE_INSTALL_FULL_SYSCONFDIR@/rdma\-ndd.conf\(ga
```

Comment:
I added documentation to the config file itself and pointed the user there.

I agree that the current configuration is quite simple but I would like to explore other directives. 

Let me think about the systemd environment variable idea more.